mitodl/open-discussions
# prosemirror editor

#### What are the relevant tickets?

closes #725 
closes #965 

#### What's this PR do?

This uses Prosemirror to implement a basic wysiwyg editor which we can use for posts and comments. The editor is basically a react wrapper (and menu) around the normal prosemirror editor. It is set-up to consume and produce Markdown, so it should drop-in to our existing use-cases without any problems. 

To start, we're just implementing a few basic editing actions (see the issue for the complete list) and if we want to get more fancy we'll come back later.

#### How should this be manually tested?

I think just play around with it and try to find bugs? Make sure that it's nice to use and all that.
